### PR TITLE
Fix #43, again (🤞)

### LINKED ISSUE #43: Make mac Window creation smoother
As of #42 the window is now "key" when it is created, meaning that it can respond to mouse move events, but on startup from the command line, there are two visual phases with a transition. The first phase shows the window but behind the terminal (at least on my machine), and the second phase pops it to the front. This is not appealing.

For most apps (ie .app bundles) the activation is handled by Launch Services, not by the app binary itself. Colin and I experimented with various things, changing the order of activation, splitting makeKeyAndOrderFront into its two separate components, etc., but weren't able to figure out the Right Way to do this.

This would be a great starter issue for somebody with low level Cocoa knowledge.

### PR DESCRIPTION
**Pull request by cmyr**
My main strategy here was to create a minimal single-window objective-c
application (based on [1]) and then to trace our executation
and compare that against the execution of the objective-c app.

There were a few places where our behaviour differed, but changing
those had no apparent effect on application behaviour; but in
the course of that work I found the actual bug, which was much sneaker.

TL;DR: We were using [NSRunningApplication activateWithOptions:],
when we should have been using [NSApplication  activateIgnoringOtherApps:].
The NSRunningApplication API is intended for controlling other running
applications; we were likely sending the msg to nil.

[1]: https://www.cocoawithlove.com/2010/09/minimalist-cocoa-programming.html

### CODE REVIEW

**Comment by ratmice** [comment]
Aha, so because the NSApplication hadn't `run()` yet when `show()` is called `[NSRunningApplication currentApplication]` returned nil, that makes a lot of sense and seems entirely plausible to me. good sleuthing!
